### PR TITLE
Disable sound on command bar buttons

### LINKED ISSUE #13730: Disable click sound on command bar buttons when they are activated by hotkey
The additional click sounds appears to be a common source of irritation on the playtest.
C&C3 avoids this problem by only playing the click sound when the button is activated using the mouse.

### PR DESCRIPTION
**Pull request by rob-v**
Closes #13730 

Add new ButtonWidget.PlaySoundOnKeyPress property and apply it (PlaySoundOnKeyPress: false) on command bar buttons to remove sound on key press.

### CODE REVIEW

**Review by abcdefg30 — APPROVED** [review]

**Comment by rob-v** [comment]
@pchote If players don't like sound on hotkey (and it happens only for hotkeys with appropriate button on screen, not all) and iirc, it is not standard for 'click sound' on key press in other games I propose to remove the 'click sound' for all buttons on key press. Or we could change the default to false and only add new PlaySoundOnKeyPress: true where really wanted.

File: OpenRA.Mods.Common/Widgets/ButtonWidget.cs [hunk 1]
```diff
@@ -28,6 +28,7 @@ public Hotkey Key
 		}
 
 		public bool DisableKeyRepeat = false;
+		public bool PlaySoundOnKeyPress = true;
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: How about `DisableKeySound` to match `DisableKeyRepeat`?  Would that be too unclear?

**Review by pchote — APPROVED** [review]
:+1: to this approach.  Just an open question about the flag name.

I originally tried to prototype a fix for this using separate `ClickSound` / `KeyPressClickSound` fields on the `ButtonWidget`, but ran into problems with having our `FieldLoader` code behave in the way I wanted with `null` vs `string.Empty`.  The approach here is much simpler.

File: OpenRA.Mods.Common/Widgets/ButtonWidget.cs (hunk 1 above)
Comment (rob-v) [inline OpenRA.Mods.Common/Widgets/ButtonWidget.cs]: Makes sense so I'll rename it (with default false).

**Comment by rob-v** [comment]
Updated - PlaySoundOnKeyPress renamed to DisableKeySound.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/bbe3cca85268f0802e06d4bd8563be3bba5c134d

08be5d3907955d8703fbdba03402429018cf7357
958cfcd378d53a7193319de0f50be60bbed9301d

**Pull request merged**